dataclasses_avroschema/pydantic/fields.py

```diff
@@ -210,6 +253,114 @@ def fake(self) -> uuid.UUID:
         return uuid.uuid5(namespace=uuid.NAMESPACE_URL, name=fake.pystr())
 
 
+class PastDateField(PydanticField):
+    """Like date, with the constraint that the value must be in the past."""
+
+    @property
+    def avro_type(self) -> typing.Dict:
+        return {
+            "type": INT,
+            "logicalType": DATE,
+            "pydantic-class": "PastDate",
+        }
+
+    def default_to_avro(self, date: PastDate) -> int:
+        return _default_to_avro_date(date)
+
+    def fake(self) -> datetime:
+        return fake.date_object()
+
+
```

Comment:
Referencing to the inheritance issue that I pointed before. You can do this:

```python
class AwareDatetimeField(fields.DatetimeField):
    """Like datetime, with the constraint that the value must have timezone info."""

    @property
    def avro_type(self) -> typing.Dict:
        return {
            "type": LONG,
            "logicalType": TIMESTAMP_MILLIS,
            "pydantic-class": "AwareDatetime",
        }

    def validate_default(self, default) -> bool:
        msg = f"Invalid default type. Default should be {datetime}"
        assert isinstance(default, datetime), msg
        return True
```

We should do the same for all the fields and at some point get rid of the `PydanticField`

Reply:
I changed the base class as you suggested. For the other pydantic fields like `NegativeIntField `the problem is a little different. If I use the `fields.IntField` as superclass, the signature of "avro_type" is incompatible with supertype `fields.IntField`.